### PR TITLE
Embed: refactor view

### PR DESCRIPTION
**Pull request by stsewd**
- Use version.is_sphinx_type to choose between mkdocs or sphinx
- Avoid nested code
- Check for multiple queries inside a loop instead of several if's
- Clean links only to the end result node, not to the whole body

Based on https://github.com/readthedocs/readthedocs.org/pull/7947

### CODE REVIEW

File: readthedocs/embed/views.py [hunk 1]
```diff
@@ -231,101 +265,101 @@ def _get_doc_content(project, version, doc):
...
+        for element in PQ(toc)('a')
+    ]
 
     if not section and headers:
         # If no section is sent, return the content of the first one
         # TODO: This will always be the full page content,
         # lets do something smarter here
         section = list(headers[0].keys())[0].lower()
 
-    if section:
```
Comment (stsewd) [inline readthedocs/embed/views.py:251]: Here only this big if was removed and the `query_obj` changed, everything else is the same.

**Comment by stsewd** [comment]
Need to rebase to master to have a better diff :D

File: readthedocs/embed/views.py [hunk 2]
```diff
@@ -143,39 +143,48 @@ def get(self, request):
...
+            url=url,
         )
 
 
 class EmbedAPI(SettingsOverrideObject):
     _default_class = EmbedAPIBase
 
 
-def do_embed(project, version, doc, section=None, path=None, url=None):
+def do_embed(*, project, version, doc=None, path=None, section=None, url=None):
```
Comment (humitos) [inline readthedocs/embed/views.py:187]: We are changing the signature of the function here. `doc` was _required_ previously and now it's not. Why it's not required anymore?

Comment (stsewd) [reply readthedocs/embed/views.py:187]: doc is the pagename, that's sphinx specific (isn't required when the project is mkdocs type as you can see below), and isn't required to get the response if path and section are given.

**Review by ericholscher — APPROVED** [review]
This looks good to me.

**Pull request merged**